### PR TITLE
cauchy stress case and dynamics 

### PR DESCRIPTION
**Pull request by Xiaojingtang1234**
Update Lagrangian formulation using initial configuration only @Xiangyu-Hu #263

### CODE REVIEW

**Comment by Xiaojingtang1234** [comment]
new test result converges
image

**Comment by Xiangyu-Hu** [comment]
@Xiaojingtang1234 the results looks ok. but ci did not pass.

File: tests/user_examples/ExtraSources/extra_sources_share/elastic_dynamics_cauchy.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		//=================================================================================================//
+		void CauchyIntegration1stHalf::initialization(size_t index_i, Real dt)
+		{
+			pos_[index_i] += vel_[index_i] * dt * 0.5;
+			F_[index_i] += dF_dt_[index_i] * dt * 0.5;
+			rho_[index_i] = rho0_ / F_[index_i].determinant();
+			Matd inverse_F_T = F_[index_i].inverse().transpose();
+			Matd almansi_strain = 0.5 * (Matd::Identity() - (F_[index_i] * F_[index_i].transpose()).inverse());
+			//obtain the first Piola-Kirchhoff stress from the  Cauchy stress 
+			stress_PK1_B_[index_i]  = F_[index_i].determinant() * elastic_solid_.StressCauchy(almansi_strain, F_[index_i], index_i) *
```
Comment (Xiangyu-Hu) [inline tests/user_examples/ExtraSources/extra_sources_share/elastic_dynamics_cauchy.cpp:23]: F_[index_i].transpose()).inverse() and determinant() have been computed twice and computation time is wasted.

**Review by Xiangyu-Hu — CHANGES_REQUESTED** [review]

File: tests/user_examples/ExtraSources/extra_sources_share/elastic_dynamics_cauchy.cpp (hunk 1 above)
Comment (Xiaojingtang1234) [reply tests/user_examples/ExtraSources/extra_sources_share/elastic_dynamics_cauchy.cpp:23]: The F_[index_i].determinant() has been marked as J;  While the (F_ * F_.transpose()).inverse()) 
is different with F_.transpose().inverse() . Therefore, i can only reduce F_.transpose() to F_T .

**Comment by Xiaojingtang1234** [comment]
Acutally yesterday local ctest all passed expect 2d-dambreak. 
image 
i checked the reason, saying position is out of expectation.
Then i rerun dambreak, it passed.
image

i have no idea maybe something wrong with the online test.

**Comment by Xiangyu-Hu** [comment]
> Acutally yesterday local ctest all passed expect 2d-dambreak. image i checked the reason, saying position is out of expectation. Then i rerun dambreak, it passed. image
> 
> i have no idea maybe something wrong with the online test.

It is OK as these case are set not to be tested.

**Comment by Xiangyu-Hu** [comment]
@Xiaojingtang1234 the 2d dambreak needs to rerun (by "Ctest --rerun-failed -V") to check if it is something wrong.

**Review by Xiangyu-Hu — APPROVED** [review]

**Pull request merged**